### DISCUSSION TITLE
Account Onboard Utility - can we update the automaticManagement attribute?

### DISCUSSION THREAD

**Discussion by Francesco-f7o** [cyberark/epv-api-scripts#236] [Ideas]
We have a RETS API capability to update the account field defining if automatic credential management is enabled or not at the account level.
In some scenarios it would be useful to be able to update this field in bulk, for example to stop or to start credential management on a specific group of accounts, when they are not belonging to the same platform, or stored in the same safe, or during a migration process, where we need to pause automatic management for a limited amount of time.
The feature is described here: https://docs.cyberark.com/Product-Doc/OnlineHelp/PAS/Latest/en/Content/SDK/UpdateAccount%20v10.htm#Bodyparameters
Is there a chance we could include the capability to update this account attribute in the account onboard utility script?

**Comment by AssafMiron** [comment]
Hello @Francesco-f7o,

Check this script and see if it fits your needs
https://github.com/cyberark/epv-api-scripts/tree/main/Get%20Accounts#invoke-bulkaccountactions

Regards,
Assaf

**Reply by Francesco-f7o** [reply]
Thanks Assaf, I tested the script, and I can successfully update other account properties, but when I try to change the automatic management I get the following output:

./Update-Account.ps1 -PVWAURL https://pvwaaddress.local/PasswordVault -AccountID 35_13 -ParameterNames "automaticManagementEnabled" -ParameterValues "True"
Account Details (before change):


categoryModificationTime  : 1646042859
id                        : 84_3
name                      : Application-LabSAP-application_target-application_user-1234-client
address                   : application_target
userName                  : application_user
platformId                : LabSAP
safeName                  : CP_RHEL
secretType                : password
platformAccountProperties : @{SystemNumber=1234; Client=client}
secretManagement          : @{automaticManagementEnabled=False; manualManagementReason=No Reason; lastModifiedTime=1626358673}
createdTime               : 1626358673

A hash table can only be added to another hash table.
At C:\temp\REST\Update-Account.ps1:244 char:6
+                     $arrProperties += $_bodyOp
+                     ~~~~~~~~~~~~~~~~~~~~~~~~~~
    + CategoryInfo          : InvalidOperation: (:) [], RuntimeException
    + FullyQualifiedErrorId : AddHashTableToNonHashTable

A hash table can only be added to another hash table.
At C:\temp\REST\Update-Account.ps1:249 char:6
+                     $arrProperties += $_bodyOp
+                     ~~~~~~~~~~~~~~~~~~~~~~~~~~
    + CategoryInfo          : InvalidOperation: (:) [], RuntimeException
    + FullyQualifiedErrorId : AddHashTableToNonHashTable

Properties that will change in Account:
Property : /secretManagement/manualManagementReason =

Account Details (after change):
categoryModificationTime  : 1646043066
id                        : 84_3
name                      : Application-LabSAP-application_target-application_user-1234-client
address                   : application_target
userName                  : application_user
platformId                : LabSAP
safeName                  : CP_RHEL
secretType                : password
platformAccountProperties : @{SystemNumber=1234; Client=client}
secretManagement          : @{automaticManagementEnabled=False; manualManagementReason=No Reason; lastModifiedTime=1626358673}
createdTime               : 1626358673

Logoff Session...
